sql/expression/function/rpad_lpad.go

```diff
@@ -0,0 +1,165 @@
+package function
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+type padType int
+
+const (
+	leftPadType  = iota
+	rightPadType
+)
+
+// MakePadder returns a Pad creator functions with a specific padType.
+func MakePadder(pType padType) func(e ...sql.Expression) (sql.Expression, error) {
+	return func(e ...sql.Expression) (sql.Expression, error) {
+		return NewPad(pType, e...)
+	}
+}
+
+// NewLogBase creates a new LogBase expression.
+func NewPad(pType padType, args ...sql.Expression) (sql.Expression, error) {
+	argLen := len(args)
+	if argLen != 3 {
+		return nil, sql.ErrInvalidArgumentNumber.New("3", argLen)
+	}
+
+	return &Pad{args[0], args[1], args[2], pType}, nil
+}
+
+// Pad is a function that pads a string with another string.
+type Pad struct {
+	str     sql.Expression
+	len     sql.Expression
```

Comment:
Since `len` is a builtin function in Go we could use another name, `length` maybe?